python/gstaichi/lang/_fast_caching/hash_utils.py

```diff
@@ -0,0 +1,11 @@
+import hashlib
+from typing import Iterable
+
+
+def hash_iterable_strings(strings: Iterable[str], separator: str = "_") -> str:
```

Comment:
Why there should be a separator? This is weird. Just hash the concatenated input no? I don't think there is any risk of things colliding because different lists reduces to the same concatenation.

Reply:
I would prefer to add a separator, and not having to think about the possiblity of adjacent things combining in ways we didn't intend.